### PR TITLE
Add latex repr to QuantumState object

### PR DESCRIPTION
**Pull request by nonhermitian**
### Summary
Adds a latex repr for pretty printing QuantumState objects (statevectors, density matrices) in Jupyter.

### CODE REVIEW

File: qiskit/quantum_info/states/quantum_state.py [hunk 1]
```diff
@@ -55,6 +56,129 @@ def __repr__(self):
         return '{}({}, dims={})'.format(
             self.rep, self.data, self._dims)
 
+    def _repr_latex_(self):
+        """
+        Generate a LaTeX representation of the Quantum State.
+
+        This file is part of QuTiP: Quantum Toolbox in Python.
```
Comment (chriseclectic) [inline qiskit/quantum_info/states/quantum_state.py]: Is this supposed to be here?

File: qiskit/quantum_info/states/quantum_state.py [hunk 2]
```diff
@@ -55,6 +56,129 @@ def __repr__(self):
...
+
+        Copyright (c) 2011 and later, Paul D. Nation and Robert J. Johansson.
+        All rights reserved.
+        """
+        shape = self._data.shape
```
Comment (chriseclectic) [inline qiskit/quantum_info/states/quantum_state.py]: This assumes that `state._data` is a numpy array, which is true for Statevector and DensityMatrix classes, but shouldn't be assumed in the base class I don't think. Can you move this to the specific classes?

File: qiskit/quantum_info/states/quantum_state.py (hunk 1 above)
Comment (nonhermitian) [reply qiskit/quantum_info/states/quantum_state.py]: Yes. Better than putting at the top.

File: qiskit/quantum_info/states/quantum_state.py (hunk 2 above)
Comment (nonhermitian) [reply qiskit/quantum_info/states/quantum_state.py]: Is there some other inputs than numpy arrays planned here?  Sparse would also be easy to handle. I can move though. Will duplicate the code.

**Comment by nonhermitian** [comment]
closing to move elsewhere

**Pull request closed without merging**